src/main/webapp/WEB-INF/static/js/trialmanager/environments.js

```diff
@@ -447,6 +447,19 @@ environmentModalConfirmationText, environmentConfirmLabel, showAlertMessage, sho
 				$scope.instanceInfo.numberOfInstances -= 1;
 
 				TrialManagerDataService.deleteInstance(index + 1);
+				updateExpermentalDesign();
+			}
+
+			function updateExpermentalDesign() {
+				studyInstanceService.getStudyInstances().then(function(instances) {
```

Comment:
As discussed, we can optimize by not having to call getStudyInstances (this is a heavy call just checking if any experiments remain for plot dataset). 

See if you can use $scope.hasExperimentalDesign or maybe check the length of the observations table (if that is reliable in the sense that if it currently loads any remaining experiments ) or if not, we can consider implementing a RESTful API for returning number of experiments a dataset has.